src/azure-cli/azure/cli/command_modules/resource/_validators.py

```diff
@@ -210,24 +211,48 @@ def validate_metadata(namespace):
         namespace.metadata = dict(x.split('=', 1) for x in namespace.metadata)
 
 
-def validate_msi(namespace):
-    if namespace.assign_identity is not None:
-        identities = namespace.assign_identity or []
-        if any(identity != MSI_LOCAL_ID for identity in identities):
-            raise CLIError("usage error: 'User assigned identities are not supported "
-                           "with --assign-identity and policy assignments'")
+def process_assign_identity_namespace(cmd, namespace):
+    validate_msi(cmd, namespace, from_identity_command=True)
+
+
+def process_assignment_create_namespace(cmd, namespace):
+    validate_msi(cmd, namespace)
+
+
+def validate_msi(cmd, namespace, from_identity_command=False):
+    identities = None
+    if from_identity_command:
+        if namespace.mi_system_assigned is not None or namespace.mi_user_assigned is None:
+            identities = [MSI_LOCAL_ID]
+            if namespace.mi_user_assigned is not None:
+                raise azclierror.AzCLIError('usage error: only one type of managed identity is allowed')
```

Comment:
Actually, I mean, could we use the more specific error type, such as `ArgumentUsageError`?
Please refer to this doc [error_handling_guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md) to throw the appropriate error type

Reply:
Sorry, fixing it now

Reply:
updated